model/project_ref.go

```diff
@@ -1762,17 +1762,18 @@ func SaveProjectPageForSection(projectId string, p *ProjectRef, section ProjectP
 			return false, errors.New("can't default project ref for a repo")
 		}
 	}
+	defaultToRepo := false
 	if p == nil {
+		defaultToRepo = true
 		p = &ProjectRef{} // use a blank project ref to default the section to repo
 	}
+
 	var err error
 	switch section {
 	case ProjectPageGeneralSection:
 		setUpdate := bson.M{
 			ProjectRefEnabledKey:                 p.Enabled,
 			ProjectRefBranchKey:                  p.Branch,
```

Comment:
Should branch default to repo? This seems like it could introduce conflicts (although I'm slightly fuzzy on the meaning of a branch when it comes to repos) cc @ablack12

Reply:
good point, especially considering that people probably specifically want different projects under one repo to correspond do different branches off of that repo.

Reply:
Discussed off the PR -- still makes sense to default since we don't know users use case